### PR TITLE
enable faction fly on join

### PR DESCRIPTION
**Pull request by minksuki**
Closes https://github.com/drtshock/Factions/issues/1206.

This adds a boolean `(default: true)` to the config that allows server administrators to determine whether or not they would like faction fly enabled when a player joins if they had it enabled before they last logged out.

### CODE REVIEW

File: src/main/java/com/massivecraft/factions/listeners/FactionsPlayerListener.java [hunk 1]
```diff
@@ -102,6 +102,11 @@ public void run() {
             P.p.log(Level.INFO, "Found %s on admin Bypass without permission on login. Disabled it for them.", player.getName());
         }
 
+        // If they last logged off with faction fly enabled, we can re-enable it so they don't fall and die
+        if (P.p.getConfig().getBoolean("f-fly.enable-flight-on-login") && me.isFlying() || me.isAutoFlying()) {
+            me.setFlying(true);
```
Comment (drtshock) [inline src/main/java/com/massivecraft/factions/listeners/FactionsPlayerListener.java:107]: Shouldn't we check if they are not on the ground?

Comment (minksuki) [reply src/main/java/com/massivecraft/factions/listeners/FactionsPlayerListener.java]: That's sort of the point of the `me.isFlying()` is it not? It'll check whether or not the player is flying and if so set the player to actually being flying.

Comment (drtshock) [reply src/main/java/com/massivecraft/factions/listeners/FactionsPlayerListener.java:107]: That checks their FPlayer if they are set to currently flying (were flying on logout). It doesn't check their Bukkit player to see if they are actually flying or not.

Comment (minksuki) [reply src/main/java/com/massivecraft/factions/listeners/FactionsPlayerListener.java]: Added a check in https://github.com/minksuki/Factions/commit/cf70de1170cc7d2e7f601cf57fb8776a11e04b2f. Thanks for the suggestion.

**Comment by CmdrKittens** [comment]
What happens if a player is flying in their faction's territory, leaves, and comes back much later to an enemy territory?

**Comment by CmdrKittens** [comment]
With unresolved concerns, I am closing this for now.

**Pull request closed without merging**